hazelcast/src/main/java/com/hazelcast/client/impl/protocol/AbstractCallableMessageTask.java

```diff
@@ -31,9 +31,16 @@ protected AbstractCallableMessageTask(ClientMessage clientMessage, Node node, Co
 
     @Override
     public final void processMessage() {
-        Object result = call();
-        endpoint.sendResponse(result, parameters.getCorrelationId());
+        try {
+            final ClientMessage result = call();
+            endpoint.sendClientMessage(result);
```

Comment:
Who sets the correlation id of this client message? How about if we make sendClientMessage getting a correlation id as parameter, and "sendClientMessage" sets the value inside the method so that we do not accidentally forget to set the correlation id.